### PR TITLE
[ISSUE 306][Flink] Support consume/produce from/to different cluster in the same process.

### PR DESCRIPTION
**Pull request by ahuazhu**
## What is the purpose of the change

Fix [ISSUE 306](https://github.com/apache/rocketmq-externals/issues/306)

When a job consumes messages from ClusterA and sends messages to ClusterB.  Because Flink operators may be chained, the consumer and producer is possible running in the same JVM process. The default instanceName creator creates the same instanceName for the producer and consumer, then the messages will be not sent to the ClusterB. 

## Brief changelog
Add setInstanceName method to Sink and Source, so the user could set the custom instanceName.

### CODE REVIEW

File: rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSink.java [hunk 1]
```diff
@@ -87,7 +94,8 @@ public void open(Configuration parameters) throws Exception {
         Validate.notNull(serializationSchema, "KeyValueSerializationSchema can not be null");
 
         producer = new DefaultMQProducer();
-        producer.setInstanceName(String.valueOf(getRuntimeContext().getIndexOfThisSubtask()));
+
+        producer.setInstanceName(this.instanceName.orElse(String.valueOf(getRuntimeContext().getIndexOfThisSubtask())));
```
Comment (vesense) [inline rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSink.java]: There are usually more than one threads/tasks in one process in the distributed environment. So you cannot use same instanceName for an operator. 
The following is a candidate:

> producer.setInstanceName(this.instanceName + String.valueOf(getRuntimeContext().getIndexOfThisSubtask()));

File: rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSource.java [hunk 2]
```diff
@@ -115,7 +125,7 @@ public void open(Configuration parameters) throws Exception {
         pullConsumerScheduleService = new MQPullConsumerScheduleService(group);
         consumer = pullConsumerScheduleService.getDefaultMQPullConsumer();
 
-        consumer.setInstanceName(String.valueOf(getRuntimeContext().getIndexOfThisSubtask()));
+        consumer.setInstanceName(this.instanceName.orElse(String.valueOf(getRuntimeContext().getIndexOfThisSubtask())));
```
Comment (vesense) [inline rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSource.java]: Same as above.

**Comment by vesense** [comment]
@ahuazhu Thanks for your contribution. I could merge this PR after comments are addressed.

File: rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSink.java (hunk 1 above)
Comment (ahuazhu) [reply rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSink.java]: Thanks. It's a simpler way to resolve the problem: create clients for each sink and source.
The PR has been updated.

**Comment by ahuazhu** [comment]
@vesense 

PR was updated. UUID was appended to the instanceName. Thus each sink/source task use an independent instance name.

**Comment by vesense** [comment]
Thanks @ahuazhu I will merge it today.

**Pull request merged**